### ISSUE TITLE
Help with youtube outlink redirect. 

### ISSUE THREAD

**Issue by Abhinav1217** [einaregilsson/Redirector#319]
_tl;dr: I need help finding a regex which target value of 'q' parameter._

I currently have a wildcard redirect for youtube outgoing link, as follows

`Redirect:` `https://www.youtube.com/redirect?event=video_description&redir_token=*&q=*&v=*`
`to:` `$2`

`Example:` `https://www.youtube.com/redirect?event=video_description&redir_token=QUFFLUhqa2FoeHZYVmhYcV9samdJTnFITkFKRm9la0t5QXxBQ3Jtc0trRmZsUmJSSDQ1R2kyV01zb3ZGM1l0eHU3ZkpON0E4aXFCWl9CZXJwTW04NkhMbHhVZUhMaEtNNEpKRjFta2NZaG9GRlpHVmpVOG8weHUwX1lwaVhsbnFncVc5X3BmUGxGVkw5dnFDNDMxS1NKMll0SQ&q=https%3A%2F%2Fwiteboard.com%2F&v=PiYS3PMXOjc`

`Process match URL Decode` 
`Output: https://witeboard.com/`

The issue is that sometimes I have noticed that there are some other parameters or a different order of parameter in youtube outgoing link. Only thing common is that parameter is called `q`. I have been trying some regex based on your double click example, some version of this.

`^(?:https?://)www.youtube.com/redirect?event=video_description&redir_token=*&q=/.*\?(http?s://.*)&v=*`

But so far I am unable to target the value of `q` paramter. The wildcard version is working most of the time unless there is a pattern mismatch. That is why I am trying to switch to regex. 
Can you help me find the regex for this?

**Comment by Gitoffthelawn** [comment]
I wrote this really fast for you, and didn't have time to test it, but it will likely work:

***Redirect:***
`^https?://www\.youtube\.com/redirect\?.*&q=([^&]*)`
***To:***
`$1`

Don't forget to select `regex` instead of `wildcard` for this redirect rule.  Depending on your web browser, selecting the `URL Decode` advanced option for this redirect rule may be required.

**Comment by Abhinav1217** [comment]
@Gitoffthelawn You my friend are a regex wizard. It is working and This should be perfect.

This should be added to  the readme file along with other patterns.

**Comment by Gitoffthelawn** [comment]
Thank you for the kind words.  I'm glad I was able to help you.

**Comment by jstairs09-cpu** [comment]
hi
